test/coverage-system/coverage.test.ts

```diff
@@ -8,114 +8,77 @@ import {Config} from "../../src/extension/config";
 
 const stubConfig = sinon.createStubInstance(Config) as Config;
 
-// Original functions
-const readFile = fs.readFile;
-const showQuickPick = vscode.window.showQuickPick;
+suite("Coverage Tests", () => {
+    teardown(() => sinon.restore());
 
-suite("Coverage Tests", function() {
-    teardown(function() {
-        (fs as any).readFile = readFile;
-        (vscode as any).window.showQuickPick = showQuickPick;
-    });
-
-    test("Constructor should setup properly @unit", function() {
+    test("Constructor should setup properly @unit", () => {
         expect(() => {
             new Coverage(stubConfig); // tslint:disable-line
         }).not.to.throw();
     });
 
-    test("#load: Should reject when readFile returns an error @unit", function(done) {
-        // tslint:disable-next-line
-        const readFile = function(path: string, cb) {
-            expect(path).to.equal("pathtofile");
-            const error: NodeJS.ErrnoException = new Error("could not read from fs");
-            return cb(error, Buffer.from(""));
-        };
-        (fs as any).readFile = readFile;
+    test("#load: Should reject when readFile returns an error @unit", async () => {
+        sinon.stub(fs, "readFile").callsFake(() => {
+            throw new Error("could not read from fs");
+        });
 
         const coverage = new Coverage(
             stubConfig,
         );
 
-        coverage.load("pathtofile")
-            .then(function() {
-                return done(new Error("Expected error did not fire!"));
-            })
-            .catch(function(error) {
-                if (error.name === "AssertionError") { return done(error); }
-                if (error.message === "could not read from fs") { return done(); }
-                return done(error);
-            });
+        try {
+            await coverage.load("pathtofile");
+        } catch (e) {
+            expect(e.message).to.equal("could not read from fs");
+        }
```

Comment:
[F] I have moved all of the tests onto async / await as I find it easier to work with. Seems like all TS is headed in this direction now. LMK what you think.

Reply:
Nice this vs the promise based approaches seem both equally "good" and match what most developers work with so I think this is fine. As long as we don't start using nested callbacks again 😁 .